homeassistant/components/sensor/influxdb.py

```diff
@@ -173,9 +173,11 @@ def update(self):
 
         points = list(self.influx.query(self.query).get_points())
         if len(points) == 0:
-            _LOGGER.error('Query returned no points : %s', self.query)
-            return
-        if len(points) > 1:
-            _LOGGER.warning('Query returned multiple points, only first one'
-                            ' shown : %s', self.query)
-        self.value = points[0].get('value')
+            _LOGGER.info('Query returned no points, sensor state set'
+                         ' to UNKNOWN : %s', self.query)
+            self.value = None
+        else:
+            if len(points) > 1:
```

Comment:
Those two lines can not be combined : the test is only for the log. The last line has to be executed for 1 and more points